mothership-ec/cog-mothership-commerce
# Update stock from barcode data

#### What does this do?

Updates stock from a text file of barcodes (which is created by the barcode scanner). This makes updating stock from a stock take a lot simpler, as we can take the dump from the scanner and run it through this task.

It takes three arguments:
- The first argument is required, and is the path to the text file. It will load this file and loop through the lines, and count how many of each identical line there is. The task is dumb, and will attempt to interpret whatever file you give it as a list of barcodes.
- The second argument is an optional switch to clear all stock data. Set it as 1 to clear all stock data within the stock location. If it is set to 0 (or anything other than 1) it will only update stock whose barcode is in the text file. Defaults to 0.
- The third argument is the stock location. The task will only update one stock location at a time. So if you run the script with a stock location of 'web', any other stock locations, such as 'bin', will remain untouched. Defaults to 'web'
#### How should this be manually tested?

Run

```
bin/cog task:run commerce:stock:from_barcodes --env=:env: '/path/to/file' :arg2: :arg3:
```

Even though this is meant for Union, I have been testing this in Uniform Wares. It shouldn't make a difference but Union already has like 5 branches merged into a compatibility branch and I couldn't bring myself to do the same.

I tested it by creating a text file with a variety of different barcodes from the UW database in a random order, and checking that the end results in the database matched the file. I tried this with both settings for the clear data argument, and with setting the location

It should also be confirmed that you can edit the stock of the affected unit after the script has been run via the admin panel
#### Related PRs / Issues / Resources?
#### Anything else to add? (Screenshots, background context, etc)
